File: round 1 code/r1d16_files/count_pos_sum_neg.py
# Name: Count of positives / sum of negatives
# Difficulty: 8 kyu
#
# --- sources ---
# Website: CodeWars
# URL: https:#www.codewars.com/kata/count-of-positives-slash-sum-of-negatives/train/javascript
#
#
#
##################
#                #
#  Instructions  #
#                #
##################

# Given an array of integers.
#
# Return an array, where the first element is the count of positives numbers and the second element is sum of negative numbers.
#
# If the input array is empty or null, return an empty array.
#
# Example
# For input [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, -11, -12, -13, -14, -15], you should return [10, -65].

##############
# PSEUDOCODE #
##############
#



#############
# SOLUTIONS #
#############

#***********#
# Attempt 1 #
#***********#

def count_positives_sum_negatives(arr):
    new_arr = []

    if len(arr) > 0:
        pos_count = 0
        neg_total = 0

        for num in arr:
            if num>0:
                pos_count += 1
            elif num<0:
                neg_total += num

        return [pos_count, neg_total]
    else:
        return []

File: round 1 code/r1d16_files/count_pos_sum_neg.js
// Name: Count of positives / sum of negatives
// Difficulty: 8 kyu
//
// --- sources ---
// Website: CodeWars
// URL: https://www.codewars.com/kata/count-of-positives-slash-sum-of-negatives/train/javascript
//
//
//
////////////////////
//                //
//  Instructions  //
//                //
////////////////////

// Given an array of integers.
//
// Return an array, where the first element is the count of positives numbers and the second element is sum of negative numbers.
//
// If the input array is empty or null, return an empty array.
//
// Example
// For input [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, -11, -12, -13, -14, -15], you should return [10, -65].

/////////////////
// PSEUDOCODE //
/////////////////

// # declare a variable count
// # declare a variable totalNeg
// # go through the array
//
// # if the number is positive (greater than 0)
//     # increment count by 1
// # if the number is negative
//     # add it to the total of negatives
//
// # return and array with the first element as the count and the second element as the sum of negatives



///////////////
// SOLUTIONS //
///////////////

//***********//
// Attempt 1 //
//***********//

function countPositivesSumNegatives(input) {
    if (input !== null && input.length > 0) {
      count = 0
      total = 0
      for(let i=0; i<input.length; i++) {
        if (input[i] > 0) {
          count += 1
        } else {
          total += input[i]
        }
      }
      return [count,total]
    } else {
      return []
    }

}
